src/common/multihash_isal.c

```diff
@@ -0,0 +1,444 @@
+/**
+ * (C) Copyright 2019-2020 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+ * The Government's rights to use, modify, reproduce, release, perform, display,
+ * or disclose this software are subject to the terms of the Apache License as
+ * provided in Contract No. B609815.
+ * Any reproduction of computer software, computer software documentation, or
+ * portions thereof marked with this legend must also reproduce the markings.
+ */
+
+#define D_LOGFAC	DD_FAC(csum)
+
+#include <stddef.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdint.h>
+
+#include <isa-l.h>
+#include <isa-l_crypto.h>
+#include <gurt/types.h>
+#include <daos/common.h>
+#include <daos/multihash.h>
+
+/**
+ * ---------------------------------------------------------------------------
+ * ISA-L Hash Algorithms
+ * ---------------------------------------------------------------------------
+ */
+
+/** CRC16_T10DIF*/
+static int
+crc16_init(void **daos_mhash_ctx)
+{
+	D_ALLOC(*daos_mhash_ctx, sizeof(uint16_t));
```

Comment:
@ryon-jensen I know that this is a bummer to allocate memory for this. The issue is that we have no proper storage via the multihash interface. The other option would be to have an interface for the user to pass a buffer and store it in the daos_mhash_ctx pointer. Would you have preferred this option instead? Any idea suggestions?

Reply:
I think this way is acceptable. I tried to reuse the same csummer as much as possible for a container on both the server and client so init is called as infrequently as possible.